### DISCUSSION TITLE
Position of id property for a CRUD route

### DISCUSSION THREAD

**Discussion by neilangell** [mockoon/mockoon#1378] [Help] [answered]
I can see you can set a custom name for the id property in a CRUD route. But is it possible to specify where the id property will be provided in the request rather than assuming it will always be at the end of the path? I have created a CRUD route with a path of users/:id/details, but when I make a GET users/23/details request it returns all users in the data bucket not just user 23. Many thanks.

**Comment by 255kb** [comment, marked as answer]
No it's not possible. The crud route will create all the usual endpoints using your path as a starting point. You will end up with `users/:id/details`,  `users/:id/details/:id`, etc.

> Mockoon is an open-source project and relies on your support. If you found this answer helpful, please consider [sponsoring us](https://github.com/sponsors/mockoon). Additionally, we're in the process of developing a [SaaS platform](https://mockoon.com/pro/) to ensure the project's long-term sustainability. Currently, our platform offers an AI assistant, data synchronization, and team collaboration features, with cloud deployments coming soon. Thank you for your support!
